eval/eval.go

```diff
@@ -352,6 +362,32 @@ func evalArrayIndexExpression(array, index object.Object) object.Object {
 	return arrayObject.Elements[idx]
 }
 
+func (s *State) applyExtension(fn object.Extension, args []object.Object) object.Object {
+	l := len(args)
+	if l < fn.MinArgs {
+		return object.Error{Value: fmt.Sprintf("<wrong number of arguments got=%d, want %s>",
+			l, fn.Inspect())} // shows usage
+	}
+	if fn.MaxArgs != -1 && l > fn.MaxArgs {
+		return object.Error{Value: fmt.Sprintf("<wrong number of arguments got=%d, want %s>",
+			l, fn.Inspect())} // shows usage
+	}
+	for i, arg := range args {
+		if i >= len(fn.ArgTypes) {
+			break
+		}
+		if fn.ArgTypes[i] == object.FLOAT && arg.Type() == object.INTEGER {
+			args[i] = object.Float{Value: float64(arg.(object.Integer).Value)}
+			continue
+		}
+		if fn.ArgTypes[i] != arg.Type() {
+			return object.Error{Value: fmt.Sprintf("<wrong type of argument got=%s, want %s>",
```

Comment:
Not related to this PR only, but do you have error without the `< >` surrounding?

I would add them via `object.Error.Error()` only

Reply:
So, object.Error aren't quite go errors, they get printed through:

https://github.com/grol-io/grol/blob/v0.33.1/object/object.go#L171
```go
func (e Error) Inspect() string { return "<err: " + e.Value + ">" }
```
which does add <err ...> around, so I can drop some of the extra in the text itself indeed

Reply:
done